lib/evmone/instructions_calls.cpp

```diff
@@ -125,15 +125,24 @@ evmc_status_code create_impl(StackTop stack, ExecutionState& state) noexcept
     if (!check_memory(state, init_code_offset, init_code_size))
         return EVMC_OUT_OF_GAS;
 
+    const auto init_code_words = num_words(static_cast<size_t>(init_code_size));
+
     auto salt = uint256{};
     if constexpr (Op == OP_CREATE2)
     {
         salt = stack.pop();
-        auto salt_cost = num_words(static_cast<size_t>(init_code_size)) * 6;
+        auto salt_cost = init_code_words * 6;
         if ((state.gas_left -= salt_cost) < 0)
             return EVMC_OUT_OF_GAS;
     }
 
+    if (state.rev >= EVMC_SHANGHAI)
```

Comment:
Does the order of the checks matter? It may be better to check this before the salt cost check.

Reply:
I've swapped the checks